dae/dae/annotation/annotate_vcf.py

```diff
@@ -27,78 +34,150 @@ def configure_argument_parser() -> argparse.ArgumentParser:
-    parser.add_argument("output", default="-", nargs="?",
-                        help="the output column file")
-
+    parser.add_argument("-r", "--region-size", default=300_000_000,
+                        type=int, help="region size to parallelize by")
+    parser.add_argument("-w", "--work-dir",
+                        help="Directory to store intermediate output files in",
+                        default="annotate_vcf_output")
+    parser.add_argument("-o", "--output",
+                        help="Filename of the output VCF result",
+                        default=None)
     CLIAnnotationContext.add_context_arguments(parser)
+    TaskGraphCli.add_arguments(parser)
     VerbosityConfiguration.set_argumnets(parser)
     return parser
 
 
-def cli(raw_args: Optional[list[str]] = None) -> None:
-    """Run command line interface for annotate_vcf tool."""
-    # FIXME:
-    # pylint: disable=too-many-locals
-    if not raw_args:
-        raw_args = sys.argv[1:]
-
-    parser = configure_argument_parser()
-    args = parser.parse_args(raw_args)
-    VerbosityConfiguration.set(args)
-    CLIAnnotationContext.register(args)
-
-    context = get_genomic_context()
-    pipeline = CLIAnnotationContext.get_pipeline(context)
-    annotation_attributes = pipeline.annotation_schema.names
-
-    in_file = VariantFile(args.input)
-
-    if args.output == "-":
-        out_file = sys.stdout
-    else:
-        # pylint: disable=consider-using-with
-        out_file = open(args.output, "wt")
-
-    # handling the header
-    header = in_file.header
-    header.add_meta(
-        "pipeline_annotation_tool", "GPF variant annotation."
-    )
-    header.add_meta(
-        "pipeline_annotation_tool", f"{' '.join(sys.argv)}")
-    for attribute in annotation_attributes:
+def update_header(variant_file, pipeline):
+    """Update a variant file's header with annotation pipeline scores."""
+    header = variant_file.header
+    header.add_meta("pipeline_annotation_tool", "GPF variant annotation.")
+    header.add_meta("pipeline_annotation_tool", f"{' '.join(sys.argv)}")
+    for attribute in pipeline.annotation_schema.names:
         description = pipeline.annotation_schema[attribute].description
         description = description.replace("\n", " ")
-        header.info.add(
-            attribute, "A", "String",
-            description)
+        header.info.add(attribute, "A", "String", description)
 
-    print(str(header), file=out_file, end="")
 
-    # handling the variants
-    with pipeline.open() as pipeline:
-        for vcf_var in in_file:
-            buffers: List[List] = [list([]) for _ in annotation_attributes]
+def annotate(input_file, region, pipeline, out_filepath):
+    """Annotate a region from a given input VCF file using a pipeline."""
+    in_file = VariantFile(input_file)
+    update_header(in_file, pipeline)
+    out_file = VariantFile(out_filepath, "w", header=in_file.header)
```

Comment:
Consider using `with` statement with https://docs.python.org/3/library/contextlib.html#contextlib.closing for managing the open `VariantFile` in_file and out_file.

Reply:
I am very hesitant to utilise `with` in this method. The majority of the code becomes indented thrice and it considerably hurts readability. I am unable to chain all context managers in a single `with` statement as `out_file` depends on `in_file` being opened.